### PR TITLE
Automated uploading of Logos

### PR DESCRIPTION
**Pull request by neefrehman**
In this PR i've added a script that will read all of the files in the `assets` folder, compare them to already uploaded files in the CMS, and then upload the new ones. I've tested it and it's all working (you might want to delete the ustwo tests I've added @kemiljk 😄 ).

The script can be run manually when any new logos are added, and (better yet) we can also set up a CI process with github actions that will run automatically it every time a new logo is comitted.

I think this is a good approach as:
1. svg files are so small that it's no big deal to add them to the repo
1. dumping the files in a folder and then committing is a lot easier than having to add them one by one in the CMS UI
2. it allows for easier collaboration (no permissions and pricing problems with a CMS)
3. most importantly, it helps decouple us from our infrastructure a bit. If we ever need to move to another database or CMS we just need to update our script to use the new API and voilá.

To do this consistently we'll need to enforce naming conventions on the svg files. Right now I'm thinking:

```shell
|-- assets
--|-- COMPANY
--|--|-- COMPANY_LOGOTYPE_COLOURSCHEME.svg
--|--|-- ...
--|-- ...
```

This currently works fine, and I'm throwing some errors when the convention is not met (which we can make more sophisticated as needed). I'm making a distinction between `logoType` and `colourScheme` as I remember Zach mentioning he had square and wide versions of many of the logos. Do you think we should reflect that naming in the CMS @kemiljk? I think it would be best to keep all variant naming lowercase too, where possible :)

### CODE REVIEW

File: scripts/upload-assets.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+const getLogosFromFs = (): FSLogoVariant[] => {
+    const ASSETS_PATH = "./assets"
+    const allCompanies = fs.readdirSync(ASSETS_PATH);
+
+    const logos = allCompanies.reduce((allLogos, currentCompany) => {
+        const companyPath = `${ASSETS_PATH}/${currentCompany}`;
+        const variantPaths = fs.readdirSync(companyPath);
+
+        const variantData = variantPaths.map(variantPath => {
+            const [/* company */, /* logoType */, colourScheme] = variantPath.replace(".svg", "").split("_");
```
Comment (neefrehman) [inline scripts/upload-assets.ts:51]: keeping the variables commented just so we know what are if we need them

**Comment by kemiljk** [comment]
Woahhhhh this is mega @neefrehman! Love it. So much more robust and like you say, decoupling from the CMS gives us lots more freedom. This implementation sounds spot on though. 

You're right about the desire to have different types of logos (logotype versus icon etc) so the UI will also need to update to reflect that eventually too. For now though, getting the whole thing working with just logotypes will get us to v0.1 with all logos in place so we can evolve it.

I've invited Zach as a contributor so he'll be able to add files eventually too. Maybe we can open source this for PR-based logo contributions eventually too 🤔.

**Comment by neefrehman** [comment]
Grand! yes lets keep the colours as types for now.

On open-sourcing, I have thought about this too and think it's a great idea. it would take a little bit of work to hide the sensitive info (`write_key`), but definitely possible!

On that note, we should also be wary of having the `read_key` visible in the plugin. Anyone will be able to inspect the javascript and see it. maybe nothing to worry about now, but do bear in mind that it's a vector for API abuse

**Comment by kemiljk** [comment]
Absolutely, I wonder if we could swap to using  a `.env`? Or is that going to be tricky to manage in our current setup?

**Pull request merged**